File: Homeworks/Data Structures/Exams/13-September-2015/First-Last-List/Java/src/PlayWithFirstLastList.java
public class PlayWithFirstLastList {
    public static void main(String[] args) {
        IFirstLastList<String> items = FirstLastListFactory.create();
        items.add("zero");
        System.out.printf("Count: %d\r\n", items.getCount());
        System.out.printf("First: %s\r\n", items.first(1).iterator().next());
        System.out.printf("Last: %s\r\n", items.last(1).iterator().next());
        System.out.printf("Min: %s\r\n", items.min(1).iterator().next());
        System.out.printf("Max: %s\r\n", items.max(1).iterator().next());

        items.clear();

        items.add("first");
        items.add("second");
        items.add("third");
        items.add("fourth");
        System.out.printf("Count: %d\r\n", items.getCount());
        System.out.printf("First: %s\r\n", items.first(1).iterator().next());
        System.out.printf("Last: %s\r\n", items.last(1).iterator().next());
        System.out.printf("Min: %s\r\n", items.min(1).iterator().next());
        System.out.printf("Max: %s\r\n", items.max(1).iterator().next());

        System.out.printf("RemoveAll('first'): %d\r\n",
                items.removeAll("first"));
        System.out.printf("RemoveAll('fourth'): %d\r\n",
                items.removeAll("fourth"));
        System.out.printf("RemoveAll('first'): %d\r\n",
                items.removeAll("first"));
        System.out.printf("Count: %d\r\n", items.getCount());
        System.out.printf("First: %s\r\n", items.first(1).iterator().next());
        System.out.printf("Last: %s\r\n", items.last(1).iterator().next());
        System.out.printf("Min: %s\r\n", items.min(1).iterator().next());
        System.out.printf("Max: %s\r\n", items.max(1).iterator().next());
    }
}

File: Data Structures/Exams/13-September-2015/First-Last-List/C#/First-Last-List/PlayWithFirstLastList.cs
using System;
using System.Linq;

using Wintellect.PowerCollections;

class PlayWithFirstLastList
{
    static void Main()
    {
        var asd = new OrderedBag<int>();
        asd.Add(5);
        asd.Add(5);
        asd.Add(5);
        asd.Add(2);
        asd.Add(2);

        //asd.RemoveAllCopies(5);

        Console.WriteLine(string.Join(", ", asd));


        return;
        var items = FirstLastListFactory.Create<string>();
        items.Add("zero");
        Console.WriteLine("Count: {0}", items.Count);
        Console.WriteLine("First: {0}", items.First(1).FirstOrDefault());
        Console.WriteLine("Last: {0}", items.Last(1).FirstOrDefault());
        Console.WriteLine("Min: {0}", items.Min(1).FirstOrDefault());
        Console.WriteLine("Max: {0}", items.Max(1).FirstOrDefault());

        items.Clear();

        items.Add("first");
        items.Add("second");
        items.Add("third");
        items.Add("fourth");
        Console.WriteLine("Count: {0}", items.Count);
        Console.WriteLine("First: {0}", items.First(1).FirstOrDefault());
        Console.WriteLine("Last: {0}", items.Last(1).FirstOrDefault());
        Console.WriteLine("Min: {0}", items.Min(1).FirstOrDefault());
        Console.WriteLine("Max: {0}", items.Max(1).FirstOrDefault());

        Console.WriteLine("RemoveAll('first'): {0}", 
            items.RemoveAll("first"));
        Console.WriteLine("RemoveAll('fourth'): {0}", 
            items.RemoveAll("fourth"));
        Console.WriteLine("RemoveAll('first'): {0}",
            items.RemoveAll("first"));
        Console.WriteLine("Count: {0}", items.Count);
        Console.WriteLine("First: {0}", items.First(1).FirstOrDefault());
        Console.WriteLine("Last: {0}", items.Last(1).FirstOrDefault());
        Console.WriteLine("Min: {0}", items.Min(1).FirstOrDefault());
        Console.WriteLine("Max: {0}", items.Max(1).FirstOrDefault());
    }
}
